### PR TITLE
fix: return NotFoundError when cluster is unknown

### LINKED ISSUE #67: Return a NotFoundError when unable to load a cluster by URL
see the implementation of `ClusterService.loadClusterByURL(ctx context.Context, clusterURL string) (*repository.Cluster, error)` which returns a `BadParameterError` when the cluster could not be found. The returned error should be a `NotFoundError` instead.

### PR DESCRIPTION
**Pull request by xcoulon**
fixes #67

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: cluster/service/cluster_service.go [hunk 1]
```diff
@@ -397,7 +397,7 @@ func (c clusterService) loadClusterByURL(ctx context.Context, clusterURL string)
 			// oops, something wrong happened, not just the cluster not found in the db
 			return nil, errs.Wrapf(err, "unable to load cluster")
 		}
-		return nil, errors.NewBadParameterError("cluster-url", fmt.Sprintf("cluster with requested url %s doesn't exist", clusterURL))
+		return nil, errors.NewNotFoundError("cluster", clusterURL)
```
Comment (nurali-techie) [inline cluster/service/cluster_service.go]: I see, `Not Found` already handled here [link](https://github.com/fabric8-services/fabric8-cluster/blob/a45d26c5309fdaf5ad86c9ee9a19bbe20e6990be/cluster/repository/cluster_repository.go#L117).  So we can avoid processing it here.

Even I see that we can optimize and can directly call `LoadClusterByURL` [link](https://github.com/fabric8-services/fabric8-cluster/blob/a45d26c5309fdaf5ad86c9ee9a19bbe20e6990be/cluster/repository/cluster_repository.go#L112) and no need of this wrapper method any more ??

Comment (xcoulon) [reply cluster/service/cluster_service.go]: right, we could use that method, indeed. But I need to check if @alexeykazakov agrees with this change, first...

**Comment by alexeykazakov** [comment]
400 is not only about an invalid syntax. See https://tools.ietf.org/html/rfc7231#section-6.5.1

> The 400 (Bad Request) status code indicates that the server cannot or
> will not process the request due to something that is perceived to be
> a client error (e.g., malformed request syntax, invalid request
> message framing, or deceptive request routing).

Also, 404 is cacheable by default.

**Comment by xcoulon** [comment]
After discussion with @alexeykazakov we decided to close this PR. It seems like we are in a grey area with regards to which error code to use when the payload parameter are invalid in the sense that they don't match any record in the database. Also, since we have other endpoints returning a `400 Bad Request` response in similar cases, we'll stick to this "pattern".

**Pull request closed without merging**